### PR TITLE
feat: Add app hierarchy filter in Grafana

### LINKED ISSUE #964: Create filters in grafana to support subgroups views of apps
## Fix

Today, its only possible to view applications monitored by Pelorus in Grafana in 2 ways:
- individually
- all together

Users want to be able to have more options of view, like:
- hierarchically
- by cluster
- etc

## Requirements

- [ ] document new behavior
- [ ] allow new behavior to be configured by any number of levels of view

### PR DESCRIPTION
**Pull request by mateusoliveira43**
- [x] I have read [Pelorus Contributing guidelines](https://github.com/dora-metrics/pelorus/blob/master/CONTRIBUTING.md)

## Linked Issues

resolves #964

## Description

Add hierarchy filter in Grafana dashboard `by_app`.

## Testing Instructions

Create Pelorus instance with
```yaml
kind: Pelorus
apiVersion: charts.pelorus.dora-metrics.io/v1alpha1
metadata:
  name: test-hierarchy-filter
  namespace: test-pelorus-operator
spec:
  exporters:
    global: {}
    instances:
      - app_name: webhook-exporter
        exporter_type: webhook
```
and run this script with the webhook endpoint
```python
import argparse
import json
import time
from urllib import request

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("url", type=str)
    arguments = parser.parse_args()

    current_timestamp = int(time.time())

    for index, app in enumerate(
        [
            "group/subgroup/application_{index},class/application_{index}",
            "group/subgroup/application_{index}",
            "group/application_{index},class/application_{index}",
            "class/application_{index}",
            "group/application_{index}",
        ]
    ):
        json_data = json.dumps(
            {
                "app": app.format(index=index),
                "image_sha": "sha256:af4092ccbfa99a3ec1ea93058fe39b8ddfd8db1c7a18081db397c50a0b8ec77d",
                "namespace": "pelorus",
                "timestamp": current_timestamp,
            }
        )

        webhook_endpoint = request.Request(
            f"{arguments.url}/pelorus/webhook", method="POST"
        )
        webhook_endpoint.add_header("User-Agent", "Pelorus-Webhook/test")
        webhook_endpoint.add_header("X-Pelorus-Event", "deploytime")
        webhook_endpoint.add_header("Content-Type", "application/json")
        request.urlopen(url=webhook_endpoint, data=json_data.encode("utf-8"))
```
You will have the following structure
```
group/subgroup/application_0
group/subgroup/application_1
group/application_2
group/application_4
class/application_0
class/application_2
class/application_3
```
Selecting in Grafana each application individually should say 1 deployment happened, selecting subgroup should say 2, selecting group should say 4 and selecting class should say 3.

### CODE REVIEW

File: charts/pelorus/templates/dashboard-sdp-byapp.yaml [hunk 1]
```diff
@@ -1137,7 +1137,7 @@ spec:
             "options": [],
             "query": "label_values(sdp:applications, app)",
             "refresh": 2,
-            "regex": "",
+            "regex": "regex": "/([^\/]+)/g",
```
Comment (mateusoliveira43) [inline charts/pelorus/templates/dashboard-sdp-byapp.yaml:1140]: TYPO

File: exporters/committime/collector_base.py [hunk 2]
```diff
@@ -124,7 +125,7 @@ def collect(self):
             commit_metric.add_metric(
                 [
                     my_metric.namespace,
-                    my_metric.name,
+                    format_app_name(my_metric.name),
```
Comment (mateusoliveira43) [inline exporters/committime/collector_base.py:128]: TODO check if I need to add this to containerimage exporter

File: exporters/provider_common/__init__.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+APP_NAME_SEPARATOR = "/"
+
+
+def format_app_name(name: str) -> str:
```
Comment (mateusoliveira43) [inline exporters/provider_common/__init__.py:4]: TODO add comma formatter

File: _test/prometheus/test.yaml [hunk 4]
```diff
@@ -142,3 +142,153 @@ tests:
...
+    exp_samples:
+    - labels: '{}'
+      value: 0
+  - expr: avg(avg_over_time(sdp:change_failure_rate:by_app{app=~".*/application_3/.*"} [5m]))
+    eval_time: 1m
+    exp_samples:
+    - labels: '{}'
+      value: 1
+
+# TODO tests with comma
```
Comment (mateusoliveira43) [inline _test/prometheus/test.yaml:294]: TODO

**Review by mpryc — DISMISSED** [review]

**Comment by mateusoliveira43** [comment]
\unhold

**Pull request merged**